src/validation.cpp

```diff
@@ -2612,7 +2611,7 @@ bool Chainstate::ConnectBlock(const CBlock& block, BlockValidationState& state,
     // doesn't invalidate pointers into the vector, and keep txsdata in scope
     // for as long as `control`.
     std::optional<CCheckQueueControl<CScriptCheck>> control;
-    if (fScriptChecks && parallel_script_checks) control.emplace(m_chainman.GetCheckQueue());
+    if (auto& queue = m_chainman.GetCheckQueue(); queue.HasThreads() && fScriptChecks) control.emplace(queue);
```

Comment:
Dumb question: Is there a reason why the order was changed here between `HasThreads()` and `fSCriptChecks`? Is it because `HasThreads()` is cheap and the line is more readible with the check next to the initializer?

Reply:
Yep, no reason other than I thought it was more readable that way.